src/withFormik.tsx

```diff
@@ -65,6 +65,12 @@ export interface WithFormikConfig<
    * throws an error object where that object keys map to corresponding value.
    */
   validate?: (values: Values, props: Props) => void | object | Promise<any>;
+
+  /**
+   * Warning function. Must return an error object or promise that
+   * throws an warning object where that object keys map to corresponding value.
+   */
+  warn?: (values: Values, props: Props) => void | object | Promise<any>;
```

Comment:
IMHO it should be also possible to compute warnings using Yup Schema by passing `warningsSchema` parameter to keep the API consistent.